eng/pipelines/templates/jobs/ci.tests.yml

```diff
@@ -19,6 +19,13 @@ parameters:
     default: true
   - name: Matrix
     type: string
+  - name: LineCoverageTarget
```

Comment:
@benbp do you think passing these as parameters is better then the current approach we are doing in go, which is to have a separate config file? On the one hand I like passing these in the yml template but on the other hand if we ever want to enforce/check this as part of local builds we will need to be able to read these values to compare.

Reply:
I would prefer the go approach so you can know during local tests whether you passed coverage, but granted it also requires custom go tooling to parse as opposed to the built-in task that was added here. I think either we should move or rewrite the go tooling into the tools repo and support it as "the model" or find a way to add local pass/fail for cpp.

Reply:
Like Go, coverage enforcement for C++ requires tools be present on the dev's machine. At this time C++ also requires `gcc` to run (which means MSVC and Clang will not work). 

We'll need to sort out how we can make coverage enforcement part of the developer's default workflow. I've started an issue here to see if we can standardize here: https://github.com/Azure/azure-sdk-for-cpp/issues/3021